docs/getting_started/basics/execute.md

```diff
@@ -1,31 +1,35 @@
-# Executing Your Workflow
+# Workflow Execution
 
-You can execute your workflow in different ways:
+A node in a workflow gets executed when its been computed successfully and outputs a result. Once a node has been executed, it will be **clean** and will not be recomputed unless it gets **dirty**. A node will be dirty whenever a plug value is modified or the workflow gets saved. Shift will use dirty propagation to identify the nodes affected by the modifications done to a node and flag them to be recomputed.
 
-## Execute All
+There are different ways a workflow can be executed, which depends on which nodes will get computed. 
 
-Click the “Execute” button (![Execute Button](../images/toolbar/button_execute.png){:width="25" height="25"}). You should see the nodes progress bars lighting up.
+## ![Execute All Button](../../images/toolbar/execute_all_default.svg){:width="25" height="25"} Execute All
 
-![Execute All](../images/execute_all.gif)
+The "Execute All" action will compute all the nodes present in the workflow. 
```

Comment:
Should we mention that will compute all the **dirty** nodes present in the workflow? Because if you execute all the nodes, and then, without clearing the nodes, you press Execute All again, it won't execute the graph because it is already computed.

Reply:
Done!